web/packages/teleterm/src/ui/ConnectMyComputer/connectMyComputerContext.tsx

```diff
@@ -25,19 +25,38 @@ import React, {
 
 import { wait } from 'shared/utils/wait';
 
-import { Attempt, makeSuccessAttempt, useAsync } from 'shared/hooks/useAsync';
+import {
+  Attempt,
+  makeEmptyAttempt,
+  makeSuccessAttempt,
+  useAsync,
+} from 'shared/hooks/useAsync';
 
 import { RootClusterUri } from 'teleterm/ui/uri';
 import { useAppContext } from 'teleterm/ui/appContextProvider';
 
+import { Server } from 'teleterm/services/tshd/types';
+
 import type {
   AgentProcessState,
   MainProcessClient,
 } from 'teleterm/mainProcess/types';
 
+export type AgentState =
+  | AgentProcessState
+  | {
+      status: 'starting';
+    }
+  | {
+      status: 'stopping';
+    };
```

Comment:
Couldn't these two statuses be managed by the agent runner? If they were, it seems like we wouldn't need `computeAgentState` because all errors would be reported by the agent state, right?